.github/workflows/pr-validation.yaml

```diff
@@ -0,0 +1,49 @@
+name: Pull Request Validation
+on:
+  pull_request:
+    types: [opened, synchronize, reopened, edited]
+    branches: [master]
+jobs:
+  # Count the number of commits in a pull request. This can be
+  # disabled by adding a line of the following form to the pull
+  # request message:
+  #
+  #    Disable-Check: commit-count
+  #
+  # The check is case-insensitive and ignores other contents on the
+  # line as well, so it is possible to add several different checks if
+  # that is necessary.
+  count_commits:
+    name: Enforce single commit pull request
+    runs-on: ubuntu-latest
+    steps:
+    - name: Checkout source
+      uses: actions/checkout@v2
+      with:
+        ref: ${{ github.event.pull_request.head.sha }}
+        fetch-depth: 0
+    - name: Dump GitHub context (for debugging)
+      env:
+        GITHUB_CONTEXT: ${{ toJSON(github) }}
+      run: |
+        echo "GITHUB_CONTEXT: $GITHUB_CONTEXT"
+    - name: Check number of commits
+      shell: bash --norc --noprofile {0}
+      # If this becomes more advanced, we might be better of using,
+      # e.g., Perl.
+      run: |
+        cat <<EOF | egrep -qsi '^disable-check:.*\<commit-count\>'
```

Comment:
Shouldn't this match any whitespace before `disable-check` as shown in the example in the comments?

Reply:
This is a generous match, yes, it only requires that the header is correct and that there is a word `commit-count`. I think this is fine since it allows matching stuff like:

```
Disable-Check: landing-gear, commit-count
```

Reply:
I meant, it matches:
```
Disable-Check: commit-count
```
but it doesn't match:

```
        Disable-Check: commit-count
```

Reply:
That is intentional since I want to follow header conventions for RFC2822.